File: backend/src/models/ConcomitantAssessment.js
const mongoose = require('mongoose');

const physicalConcomitantsSchema = new mongoose.Schema({
  score: { type: Number, default: 0 },
  candidateEvents: { type: Number, default: 0 },
  confidence: { type: Number, default: 0 },
}, { _id: false });

const concomitantAssessmentSchema = new mongoose.Schema({
  childId: { type: mongoose.Schema.Types.ObjectId, ref: 'Child', required: true },
  speechAssessmentId: { type: mongoose.Schema.Types.ObjectId, ref: 'SpeechAssessment' }, // Comp 1 link
  taskType: { type: String, required: true },
  audioUrl: { type: String },
  videoUrl: { type: String },
  duration: { type: Number },
  analysisStatus: { type: String, enum: ['pending', 'processing', 'completed', 'failed'], default: 'completed' },
  
  physicalConcomitants: {
    facialTension: { type: physicalConcomitantsSchema, default: () => ({}) },
    eyeMovements: { type: physicalConcomitantsSchema, default: () => ({}) },
    headMovements: { type: physicalConcomitantsSchema, default: () => ({}) },
    extremityMovements: { type: physicalConcomitantsSchema, default: () => ({}) }
  },

  vocalAvoidanceIndicators: {
    responseLatency: { 
      baseline: Number,
      elevated: Boolean,
      timestamps: [Number]
    },
    wordSubstitution: [{
      originalWord: String,
      detectedAlternative: String,
      confidence: Number,
      timestamp: Number
    }],
    circumlocution: [{
      utterance: String,
      confidence: Number,
      timestamp: Number
    }],
    acousticTension: {
      summary: String,
      candidateTimestamps: [Number],
      confidence: Number
    }
  },

  eventTimeline: [{
    timestamp: Number,
    eventType: String,
    source: String, // 'Component 1' or 'Component 2'
    details: String,
    confidence: Number
  }],

  modelConfidence: { type: Number, default: 0 },
  
  clinicianVerification: {
    status: { type: String, enum: ['Pending', 'Confirmed', 'Needs Review', 'Rejected'], default: 'Pending' },
    notes: String,
    verifiedAt: Date,
    verifiedBy: { type: mongoose.Schema.Types.ObjectId, ref: 'User' }
  },

  isPrototype: { type: Boolean, default: true },
}, { timestamps: true });

module.exports = mongoose.model('ConcomitantAssessment', concomitantAssessmentSchema);

File: mobile/lib/features/concomitant/data/models/concomitant_assessment.dart
class ConcomitantAssessment {
  final String id;
  final String childId;
  final String? speechAssessmentId;
  final String taskType;
  final int duration;
  final String analysisStatus;
  final Map<String, dynamic> physicalConcomitants;
  final Map<String, dynamic> vocalAvoidanceIndicators;
  final List<dynamic> eventTimeline;
  final double modelConfidence;
  final Map<String, dynamic> clinicianVerification;
  final DateTime createdAt;

  ConcomitantAssessment({
    required this.id,
    required this.childId,
    this.speechAssessmentId,
    required this.taskType,
    required this.duration,
    required this.analysisStatus,
    required this.physicalConcomitants,
    required this.vocalAvoidanceIndicators,
    required this.eventTimeline,
    required this.modelConfidence,
    required this.clinicianVerification,
    required this.createdAt,
  });

  factory ConcomitantAssessment.fromJson(Map<String, dynamic> json) {
    return ConcomitantAssessment(
      id: json['_id'] ?? '',
      childId: json['childId'] is Map ? json['childId']['_id'] : (json['childId'] ?? ''),
      speechAssessmentId: json['speechAssessmentId'] is Map ? json['speechAssessmentId']['_id'] : json['speechAssessmentId'],
      taskType: json['taskType'] ?? 'Unknown',
      duration: json['duration'] ?? 0,
      analysisStatus: json['analysisStatus'] ?? 'pending',
      physicalConcomitants: json['physicalConcomitants'] ?? {},
      vocalAvoidanceIndicators: json['vocalAvoidanceIndicators'] ?? {},
      eventTimeline: json['eventTimeline'] ?? [],
      modelConfidence: (json['modelConfidence'] as num?)?.toDouble() ?? 0.0,
      clinicianVerification: json['clinicianVerification'] ?? {},
      createdAt: json['createdAt'] != null ? DateTime.parse(json['createdAt']) : DateTime.now(),
    );
  }
}
